AirCasting/SessionViews/ShareSession/GenerateSessionFileController.swift

```diff
@@ -0,0 +1,71 @@
+// Created by Lunar on 12/01/2022.
+//
+
+import Foundation
+import SSZipArchive
+
+protocol GenerateSessionFileController {
+    func generateFile(for session: SessionEntity) -> Result<URL, Error>
+}
+
+struct DefaultGenerateSessionFileController: GenerateSessionFileController {
+    let fileGenerator: CSVFileGenerator
+    
+    init() {
+        fileGenerator = DefaultCSVFileGenerator()
+    }
+    
+    func generateFile(for session: SessionEntity) -> Result<URL, Error> {
+        let fileContent = prepareFileContent(for: session)
+        let fileName = session.name?.replacingOccurrences(of: " ", with: "_") ?? "session"
+        let fileGenerationResult = fileGenerator.generateFile(content: fileContent, fileName: fileName)
+        
+        switch fileGenerationResult {
+        case .success(let url):
+            let zipResult = zipFile(url, fileName: fileName)
+            return zipResult
+        case .failure(_):
+            return fileGenerationResult
+        }
+    }
+    
+    private func prepareFileContent(for session: SessionEntity) -> String {
+        var content = ""
+        let headers1 = "sensor_model, sensor_package, sensor_capability, sensor_units\n"
+        let headers2 = "Timestamp, Value\n"
+        session.allStreams?.forEach({ stream in
+            content.append(headers1)
+            content.append("\(stream.sensorName ?? ""), \(stream.sensorPackageName ?? ""), \(stream.measurementType ?? ""), \(stream.unitName ?? "")\n")
+            content.append(headers2)
+            stream.allMeasurements?.forEach({ measurement in
+                if let time = measurement.time {
+                    content.append("\(time), \(measurement.value)\n")
+                }
+            })
+        })
+        return content
+    }
+    
+    private func zipFile(_ url: URL, fileName: String) -> Result<URL, Error> {
+        var newUrl = url
+        let fileManager = FileManager.default
+        do {
+            let path = try fileManager.url(for: .documentDirectory, in: .allDomainsMask, appropriateFor: nil, create: false)
+            let zipPath = path.appendingPathComponent(fileName + ".zip").path
+            SSZipArchive.createZipFile(atPath: zipPath, withContentsOfDirectory: url.path, keepParentDirectory: false)
```

Comment:
we should create a wrapper around `SSZipArchive`, make a protocol for it and inject it into this class as a protocol
```Swift
protocol FileZipper {
    func createZipFile(atPath: .... )
}

import SSZipArchive

class SSZipFileZipper: FileZipper {
    func createZipFile(atPath: .... ) {
        SSZipArchive.createZipArchive(atPath:...)
    }
}
```

this is a common practice for handling 3rd party utility libraries. Some more important reasons:
1) Easier to test - when we'll be writing unit tests for this module it will be much easier as we'd only need to check if this module is sending correct data to our stubbed out implementation of FileZipper. With current code you cannot stub it out at all.
2) If we use this lib for more places in the app (I find that possible) and this library changes its interface we'd only have to adjust one place in the app, not each usage.

On that note it's also common and beneficial to wrap `FileManager` for testing purposes. Let's skip that for now as we don't test our code, but if we ever decide to put automated unit tests in for this class we're gonna need to abstract it away.